core/src/snapshot.rs

```diff
@@ -138,7 +128,7 @@ impl SnapshotMaker {
 pub fn try_read_snapshot(
     store_dir: impl AsRef<Path>,
     kura: &Arc<Kura>,
-    query_handle: LiveQueryStoreHandle,
+    live_query_store_lazy: impl FnOnce() -> LiveQueryStoreHandle,
```

Comment:
Why closure?

Reply:
Because there are possible early returns before any use of this value.